packages/playground/src/components/ssh_keys/SshDataDialog.vue

```diff
@@ -95,16 +99,39 @@ export default defineComponent({
   setup(props, ctx) {
     const currentKey = ref<SSHKeyData>(props.selectedKey);
     const loading = ref<boolean>(false);
-
+    const hasChanges = ref<boolean>(true);
+    const originalKey = ref<SSHKeyData>({ ...props.selectedKey });
+    const showTooltip = computed(() => !hasChanges.value);
     watch(
       () => props.open,
       newValue => {
         if (newValue) {
           currentKey.value = { ...props.selectedKey };
+          originalKey.value = { ...props.selectedKey };
           loading.value = false;
         }
       },
     );
+    // watch currentKey to detect changes
+    watch(
+      () => currentKey.value,
+      (newValue, oldValue) => {
+        if (
+          oldValue.name.length &&
+          oldValue.fingerPrint &&
+          oldValue.publicKey &&
+          (newValue.name !== originalKey.value.name ||
+            (oldValue.publicKey.length && newValue.publicKey !== originalKey.value.publicKey) ||
```

Comment:
Why do you check `oldValue.publicKey.length` and `oldValue.fingerPrint.length` inside the condition ? I see it's unnecessary